### DISCUSSION TITLE
can I reload a store?

### DISCUSSION THREAD

**Discussion by patricijo** [pmndrs/zustand#1696] [Q&A]
Hello, 
I love zustand so far but I hate the rules of the EU... In europe we are requiered to ask for permession before using something like localstorage. So I am not sure how to handle it when I want to use zustand persist. Is there a function which reloades the store when cookies get acceptet? Or how you handle persistant storage?   

I am using next 13
My current store looks like this and works fine but the storage only gets persistant after full reload:

```
import { persist } from "zustand/middleware";
import { create } from "zustand";
import { getCookieConsentValue } from "react-cookie-consent";

const store = (set, get) => ({
...
});

const consent = getCookieConsentValue();
let per;
if (consent) {
  per = persist(store, {
    name: "storage",
  });
} else {
  (per = store),
    {
      name: "storage",
    };
}

export const myStore = create(per);
```

**Comment by dai-shi** [comment]
Does this help? https://github.com/pmndrs/zustand/blob/962f83e01599257aa6eceaaa993965a0e296d359/docs/guides/how-to-reset-state.md

**Reply by patricijo** [reply]
thanks for your answer. I dont think so it only resets the state not the store.

**Reply by dai-shi** [reply]
Oh, I misunderstood. Your solution seems good then.

**Comment by livwvil** [comment]
I faced with a similar problem
maybe you should use context, it is my saviour
https://codesandbox.io/s/small-moon-83s78k?file=/src/App.js
